scripts/build.py

```diff
@@ -554,37 +555,52 @@ def download_sdk(url, targetfolder, strip_components=1):
             # Android SDK
             download_sdk('%s/%s-%s-android-29-29.0.3.tar.gz' % (self.package_path, PACKAGES_ANDROID_SDK, host), join(sdkfolder, PACKAGES_ANDROID_SDK))
 
+
+    def get_ems_dir(self):
+        return join(self.ext, 'SDKs', 'emsdk-' + EMSCRIPTEN_VERSION_STR)
+
     def _form_ems_path(self):
-        path = join(self.ext, EMSCRIPTEN_DIR)
-        return path
+        upstream = join(self.get_ems_dir(), 'upstream', 'emscripten')
+        if os.path.exists(upstream):
+            return upstream
+        return join(self.get_ems_dir(), 'fastcomp', 'emscripten')
 
     def install_ems(self):
-        url = '%s/%s-%s.tar.gz' % (self.package_path, PACKAGES_EMSCRIPTEN_SDK, self.host)
-        dlpath = self._download(url)
-        if dlpath is None:
-            print("Error. Could not download %s" % url)
-            sys.exit(1)
-        self._extract(dlpath, self.ext)
-        self.activate_ems()
+        # TODO: should eventually be moved to install_sdk
+        emsDir = self.get_ems_dir()
+
         os.environ['EMSCRIPTEN'] = self._form_ems_path()
+        os.environ['EM_CONFIG'] = join(self.get_ems_dir(), '.emscripten')
+        os.environ['EM_CACHE'] = join(self.get_ems_dir(), 'emscripten_cache')
```

Comment:
Now _all_ emscripten files are located within `tmp/dynamo_home/ext/SDKs/emsdk-1.39.16` and are easy to remove and reinstall.